From b443e08165aa580ac7503eb702264c437b518617 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 17 Nov 2016 12:51:48 +0200
Subject: [PATCH 0585/1345] dts: a8040: Add Device Tree for MACCHIATOBin board

commit  0df9072f349acfe3adf8f98c7509c28439a31c73 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Add DTS file for the community board MACCHIATOBin based
on Marvell A8040 SoC.

Change-Id: I7654ee986e7262b489de9367698e6f811aa9a422
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/33880
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../bindings/arm/marvell/armada-7k-8k.txt          |    3 +
 arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts  |  100 ++++++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-8040.dtsi       |    3 +
 .../boot/dts/marvell/armada-cp110-master.dtsi      |   30 ++++++
 4 files changed, 136 insertions(+)

diff --git a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
index 60a15aa..4e722f4 100644
--- a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
@@ -38,6 +38,9 @@ Boards:
    - "marvell,armada8040-db-router", "marvell,armada8040-db"
       when the board being used is the Armada 8040 Development board
 
+   - "marvell,armada8040-mcbin"
+      when the board being used is the MACCHIATOBin community board
+
 Example:
 
 compatible = "marvell,armada7040-db-router", "marvell,armada7040-db-nand", "marvell,armada7040-db-all-pcie", "marvell,armada7040-pd",
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
index f7bb0cc..123cd44 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
@@ -56,6 +56,86 @@
 		reg = <0x0 0x0 0x0 0x80000000>;
 	};
 
+	ap806 {
+		config-space@f0000000 {
+			vccq_mmc_reg: regulator@0 {
+				compatible = "regulator-fixed";
+				regulator-name = "vccq-mmc-reg";
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+				status = "okay";
+			};
+		};
+	};
+
+	cp110-master {
+		config-space@f2000000 {
+			pinctrl@440000 {
+				xhci0_vbus_pins: xhci0-vbus-pins {
+					marvell,pins = "mpp47";
+					marvell,function = "gpio";
+				};
+				pcie_reset_pins: pcie-reset-pins {
+					marvell,pins = "mpp52";
+					marvell,function = "gpio";
+				};
+			};
+
+			gop {
+				emac0_0: mac0 {
+					phy-mode = "kr"; /* lane-4 */
+				};
+			};
+			ppv22@000000 {
+				pinctrl-names = "default";
+				eth0_0: eth0@010000 {
+					status = "okay";
+				};
+			};
+		};
+		pcie@0x600000 {
+			num-lanes = <4>;
+			status = "okay";
+		};
+		vccq_sdio_reg: regulator@1 {
+			compatible = "regulator-fixed";
+			regulator-name = "vccq-sdio-reg";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			regulator-always-on;
+			status = "okay";
+		};
+	};
+
+	cp110-slave {
+		config-space@f4000000 {
+			gop {
+				emac0_1: mac0 {
+					phy-mode = "kr"; /* lane-4 */
+				};
+				emac2_1: mac2 {
+					phy-mode = "sgmii"; /* lane-0 */
+				};
+				emac3_1: mac3 {
+					phy-mode = "sgmii"; /* lane-5 */
+				};
+			};
+			ppv22@000000 {
+				pinctrl-names = "default";
+				eth0_1: eth0@010000 {
+					status = "okay";
+				};
+				eth1_1: eth1@020000 {
+					status = "okay";
+				};
+				eth2_1: eth2@030000 {
+					status = "okay";
+				};
+			};
+		};
+	};
+
 	/* Regulator labels correspond with schematics */
 	v_3_3: regulator-3-3v {
 		compatible = "regulator-fixed";
@@ -97,6 +177,8 @@
 
 &cpm_i2c0 {
 	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&cpm_i2c0_pins>;
 	status = "okay";
 };
 
@@ -136,3 +218,21 @@
 	usb-phy = <&usb3h0_phy>;
 	status = "okay";
 };
+
+&ap_sdhci0 {
+	vqmmc-supply = <&vccq_mmc_reg>;
+	bus-width = <8>;
+	no-1-8-v;
+	non-removable;
+	xenon,phy-slow-mode;
+	status = "okay";
+};
+
+&cpm_sdhci0 {
+	vqmmc-supply = <&vccq_sdio_reg>;
+	bus-width = <4>;
+	broken-cd;
+	pinctrl-names = "default";
+	pinctrl-0 = <&cpm_sdhci_pins>;
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040.dtsi b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
index 27695a0..7e4087a 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
@@ -54,6 +54,9 @@
 	compatible = "marvell,armada8040", "marvell,armada-ap806-quad",
 		     "marvell,armada-ap806";
 	aliases {
+		gpio0 = &gpio0;
+		gpio1 = &cpm_gpio0;
+		gpio2 = &cpm_gpio1;
 		ethernet0 = &emac0;
 		ethernet1 = &emac2;
 		ethernet2 = &emac3;
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi
index 64939af..babdb7a 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi
@@ -242,6 +242,36 @@
 				clocks = <&cpm_syscon0 1 26>;
 				status = "disabled";
 			};
+
+			cpm_gpio0: gpio@440100 {
+				compatible = "marvell,orion-gpio";
+				reg = <0x440100 0x40>;
+				ngpios = <32>;
+				gpiobase = <20>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				interrupt-controller;
+				#interrupt-cells = <2>;
+				interrupts = <ICU_GRP_NSR 86 IRQ_TYPE_LEVEL_HIGH>,
+					   <ICU_GRP_NSR 85 IRQ_TYPE_LEVEL_HIGH>,
+					   <ICU_GRP_NSR 84 IRQ_TYPE_LEVEL_HIGH>,
+					   <ICU_GRP_NSR 83 IRQ_TYPE_LEVEL_HIGH>;
+			};
+
+			cpm_gpio1: gpio@440140 {
+				compatible = "marvell,orion-gpio";
+				reg = <0x440140 0x40>;
+				ngpios = <31>;
+				gpiobase = <52>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				interrupt-controller;
+				#interrupt-cells = <2>;
+				interrupts = <ICU_GRP_NSR 82 IRQ_TYPE_LEVEL_HIGH>,
+					   <ICU_GRP_NSR 81 IRQ_TYPE_LEVEL_HIGH>,
+					   <ICU_GRP_NSR 80 IRQ_TYPE_LEVEL_HIGH>,
+					   <ICU_GRP_NSR 79 IRQ_TYPE_LEVEL_HIGH>;
+			};
 		};
 
 		cpm_pcie0: pcie@f2600000 {
-- 
1.7.9.5

